### ISSUE TITLE
Parser: can't handle `ticat cmd <arg start with .>`

### ISSUE THREAD

**Issue by innerr** [innerr/ticat#14]
When an arg value has prefix `.` or `/`:
```bash
$> ticat hub.add.local ./mydir
[CmdParser.parse] hub.add.local: unknow input 'mydir', shoud be sub cmd
```
Expected: call command `hub.add.local` with arg `./mydir`

Workaround for now is calling command with arg-name:
```bash
$> ticat hub.add.local path=./mydir
```

Those bugs are due to the shitty implementation of `pkg/parser`, it should be re-write.

**Comment by tisonkun** [comment]
I can see it is because `.` and `/` are meta character as `cmdAlterSeps`. If we are gonna reimplement parser (`pkg/cli/parser`), where is the spec? I might take some time to investigate but a full feature spec is desirable. Otherwise it is hard to know what should be supported.

**Comment by innerr** [comment]
Here is the spec draft https://github.com/innerr/ticat/tree/main/doc/spec
Might missed something.

**Comment by tisonkun** [comment]
@innerr the spec tells details on each sub command. For implement a parser, the grammar is more significant. For example

```
cmd := 'ticat' commandList paramList
commandList := command [':' command]+
command := abbr[.abbr]+
abbr := \w+
paramList := ...
```

**Comment by innerr** [comment]
@tisonkun you mean a BNF describtion about the grammar, maybe you could write one? (that could use for yacc later)

**Comment by tisonkun** [comment]
@innerr make sense. We can continue discussion on a basic grammar definition.

**Comment by innerr** [comment]
Related PR: #33

**Comment by innerr** [comment]
The gramma become more clear now:
- Sequence is the most important `ticat cmd : cmd : ...`
- Env setting `ticat {key=val} cmd` could be removed and instead by commands `ticat env.set key val : <cmd>`
    * This removes lots of complications of parser
    * The key-val life span specification will lose by removing `{key=val}` gramma
        - For now we have env key-val life span:
            * Global span `ticat {key=val} : cmd`
            * Cmd level span `ticat cmd : {key=val} cmd`, affected in this cmd and it's sub flows
-

**Comment by innerr** [comment]
Fixed at #235
